### PR TITLE
Foorm: Switch to new system for CSF Intro

### PR DESCRIPTION
**Pull request by molly-moen**
Switch all CSF Intro post survey links and summaries to use Foorm for workshops starting on or after 5/8. 

Links updates are below. All links will still go to the pegasus surveys/views for workshops before 5/8.
- teacher exit survey url (linked from professional development page and sent via mailer).
- summary view linked from workshop dashboard and workshop details page

We can take out the date logic for which survey a teacher sees in a few weeks, but for now it will ensure if they are taking their survey late it won't be in a different system from the rest of the results for that workshop.

Also added a notice below each rollup table to indicate averages are from May onwards, with a link to the legacy page:
Screen Shot 2020-05-07 at 10 05 11 AM

## Links

- [jira](https://codedotorg.atlassian.net/browse/PLC-878)

## Testing story
Added a unit test and new mailer preview for exit survey logic. Manually tested links by setting up a workshop for May 9th locally and validating links all go to the new system, and testing an old workshop still showed the old links.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by breville — APPROVED** [review]
This looks great.  Thank you for the attention to detail as this comes in for landing!

File: apps/src/code-studio/pd/workshop_dashboard/workshop_dashboard.jsx [hunk 1]
```diff
@@ -124,7 +124,7 @@ export default class WorkshopDashboard extends React.Component {
               component={DailySurveyResultsLoader}
             />
             <Route
-              path="daily_survey_results_foorm(/:workshopId)"
```
Comment (bencodeorg) [inline apps/src/code-studio/pd/workshop_dashboard/workshop_dashboard.jsx:127]: Was this route name unused before this PR?

File: dashboard/app/models/pd/enrollment.rb [hunk 2]
```diff
@@ -173,12 +173,18 @@ def self.filter_for_survey_completion(enrollments, select_completed = true)
     new_academic_year_enrollments, other_enrollments = other_enrollments.partition do |enrollment|
       [Pd::Workshop::COURSE_CSP, Pd::Workshop::COURSE_CSD].include?(enrollment.workshop.course) && enrollment.workshop.workshop_starting_date > Date.new(2018, 8, 1)
     end
+    csf_intro_foorm_enrollments, other_enrollments = other_enrollments.partition do |enrollment|
```
Comment (bencodeorg) [inline dashboard/app/models/pd/enrollment.rb:176]: TIL about `partition`!

File: dashboard/app/models/pd/enrollment.rb [hunk 3]
```diff
@@ -173,12 +173,18 @@ def self.filter_for_survey_completion(enrollments, select_completed = true)
...
+        enrollment.workshop.workshop_ending_date > Date.new(2020, 5, 8)
+    end
 
     (
       filter_for_regular_survey_completion(other_enrollments, select_completed) +
       filter_for_teachercon_survey_completion(teachercon_enrollments, select_completed) +
       filter_for_local_summer_survey_completion(local_summer_enrollments, select_completed) +
-      filter_for_academic_year_survey_completion(new_academic_year_enrollments, select_completed)
+      filter_for_academic_year_survey_completion(new_academic_year_enrollments, select_completed) +
+      filter_for_csf_intro_foorm_survey_completion(csf_intro_foorm_enrollments, select_completed)
```
Comment (bencodeorg) [inline dashboard/app/models/pd/enrollment.rb:187]: 😭 at how many versions of this there are, 🙏  `foorm` replaces them all!

**Review by bencodeorg — APPROVED** [review]
It's happening!!!

File: apps/src/code-studio/pd/workshop_dashboard/workshop_dashboard.jsx (hunk 1 above)
Comment (molly-moen) [reply apps/src/code-studio/pd/workshop_dashboard/workshop_dashboard.jsx:127]: it was what we used for testing, I wanted to remove 'foorm' from the end user experience so I made the url more generic

**Pull request merged**